In [16]:
#import statements
import pandas as pd 
import numpy as np 
import pickle
import datetime
from sklearn.preprocessing import OneHotEncoder
pd.set_option('display.max_columns', None)

In [6]:
#read in the pickled dataset from the processing script:
infile = open("../processing/housedat.pickle",'rb')
dat = pickle.load(infile)
infile.close()
#list(dat.columns)

We will now start processing the data necessary for modeling    

### Training / Test Split

First, we will split the data into training and test sets:   
The training dataset contains all housing prices that were sold during the years of 2010 - 2019.    
The test dataset contains all housing prices that were sold during 2020 and 2021.   

In [7]:
dat['sale_date'] = pd.to_datetime(dat['sale_date'])
#first subset the data only to 2010 - 2021
dat = dat[dat['sale_date'].dt.year > 2009]
#split to the training set 
train = dat[dat['sale_date'].dt.year < 2020]
#split to the test set 
test = dat[dat['sale_date'].dt.year > 2019]

In [11]:
train.shape

(198168, 27)

In [12]:
test.shape

(37500, 27)

In [14]:
100 * (37500 / (37500 + 198168))

15.912215489587044

The test data represents ~16% and the training data represents ~84% of the split.  

### One Hot Encoding